### PR TITLE
feat(CX-3806): add artists filtering input to my collection

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [CX-3806] 

### Description

This PR adds the search input to the artists only view inside my collection

https://github.com/artsy/eigen/assets/11945712/6e61150b-7eff-404e-8e3f-37aa2befc266

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add artists filtering input to my collection -mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-3806]: https://artsyproduct.atlassian.net/browse/CX-3806?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/Components/MyCollectionArtistFiltersStickyTab.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { useEffect, useState } from "react"
+
+export const MyCollectionArtistFilters: React.FC<{}> = () => {
+  const space = useSpace()
+  const keyword = MyCollectionArtworksKeywordStore.useStoreState((state) => state.keyword)
+  const [query, setQuery] = useState(keyword)
+
+  const setKeyword = debounce(
+    MyCollectionArtworksKeywordStore.useStoreActions((actions) => actions.setKeyword)
+  )
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Components/MyCollectionArtistFiltersStickyTab.tsx:14]: Don't we need to pass a delay to `debaunce`?

```suggestion
  const setKeyword = debounce(
    MyCollectionArtworksKeywordStore.useStoreActions((actions) => actions.setKeyword)
  )
```

File: src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tsx [hunk 2]
```diff
@@ -63,6 +52,24 @@ export const MyCollectionCollectedArtistsView: React.FC<MyCollectionCollectedArt
...
+          .normalize("NFD")
+          .replace(/[\u0300-\u036f]/g, "")
+          .includes(
+            keyword
+              .toLowerCase()
+              .trim()
+              .normalize("NFD")
+              .replace(/[\u0300-\u036f]/g, "")
+          )
+      )
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tsx:71]: **suggestion:x** Maybe we can move this into a function.

File: src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tsx [hunk 3]
```diff
@@ -72,64 +79,37 @@ export const MyCollectionCollectedArtistsView: React.FC<MyCollectionCollectedArt
...
+          )
+        }}
+        onEndReached={handleLoadMore}
+        ListFooterComponent={!!hasNext ? <LoadingIndicator /> : <Spacer y={2} />}
+        ItemSeparatorComponent={() => <Spacer y={2} />}
+        refreshControl={
+          !__TEST__ ? (
+            <RefreshControl onRefresh={handleRefresh} refreshing={refreshing} />
+          ) : undefined
+        }
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tsx:110]: **suggestion:** You might want to use our hook [useRefreshControl](https://github.com/artsy/eigen/blob/e1a222b6dac24fd0ff269b6ac7f09af00d49fcf7/src/app/utils/refreshHelpers.tsx#L33) here.

Comment (olerichter00) [reply src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tsx:110]: **question:** Is there a reason for the `!__TEST__`?

**Review by olerichter00 — APPROVED** [review]
Great work 🌟

File: src/app/Scenes/MyCollection/Components/MyCollectionArtistFiltersStickyTab.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionArtistFiltersStickyTab.tsx:14]: the default time was working well that's why I didn't add. but it only makes sense to add it for consistency. I will update

File: src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tsx (hunk 2 above)
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tsx:71]: good idea - I was actually thinking about that 😄. Will do

File: src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tsx (hunk 3 above)
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tsx:110]: thanks for sharing this! i didn't see it before

File: src/app/Scenes/MyCollection/Components/MyCollectionArtistFiltersStickyTab.tsx (hunk 1 above)
Comment (olerichter00) [reply src/app/Scenes/MyCollection/Components/MyCollectionArtistFiltersStickyTab.tsx:14]: Up to you whether you want to use the default or pass a delay. :)

**Pull request merged**